apps/studio/src/features/sign-in/components/SingpassLogin/SingpassLoginButton.tsx

```diff
@@ -0,0 +1,46 @@
+import { useRouter } from "next/router"
+import { Box, Flex, Stack, Text } from "@chakra-ui/react"
+import { Button } from "@opengovsg/design-system-react"
+
+import { SingpassFullLogo } from "~/components/Svg/SingpassFullLogo"
+import { trpc } from "~/utils/trpc"
+import { getRedirectUrl } from "~/utils/url"
+
+export const SingpassLoginButton = (): JSX.Element | null => {
+  const router = useRouter()
+  const singpassLoginMutation = trpc.auth.singpass.login.useMutation({
+    onSuccess: async ({ redirectUrl }) => {
+      await router.push(redirectUrl)
```

Comment:
why `push` over `replace` here? was thinking that the previous url will be the singpass callback, so shouldn't be in the history stack

Reply:
Hmm the URL would be `/singpass/login` which shows the screen to start the authentication process with Singpass. If we were to do a `replace`, the back button will bring the user back to the original email login page which would not be intended.

I think it should be possible for the user to come back to the `/singpass/login` page via the back button and they can re-initiate the Singpass login flow again using the same button.

Reply:
ah ok, so the previous page ise the page iwth teh singpass authenticate button! sounds good then, thought it was the callback initiated by singpass to our app